src/thunderbots/software/ai/passing/evaluation.cpp

```diff
@@ -0,0 +1,52 @@
+// TODO: start of file comment
+
+#include "ai/passing/evaluation.h"
+#include "util/parameter/dynamic_parameters.h"
+
+double getStaticPositionQuality(Field field, Point position) {
+
+    double positionQuality = 1;
+    double length          = field.length() / 2;
+    double width           = field.width() / 2;
+
+    // This constant is used to determine how steep the sigmoid slopes below are
+    static const double sigmoid_steepness = 15;
+
+    // The offset from the sides of the field for the center of the sigmoid functions
+    // TODO: should we be using dynamic parameters like this here?
+    double x_offset = Util::DynamicParameters::AI::Passing::static_position_quality_x_offset.value();
+    double y_offset = Util::DynamicParameters::AI::Passing::static_position_quality_y_offset.value();
+    double goal_weight = Util::DynamicParameters::AI::Passing::static_position_quality_friendly_goal_distance_weight.value();
+
+    if (position.x() >= 0)
+    {
+        // Positive x is closer to the enemy goal, so the higher the better!
+        positionQuality =
+                positionQuality / (1 + std::exp(sigmoid_steepness * (position.x() - (length - x_offset))));
+    }
+    else if (position.x() < 0)
+    {
+        // Negative x is closer to our goal, so the lower the worse it is
+        positionQuality =
+                positionQuality / (1 + std::exp(sigmoid_steepness * (-position.x() - (length - x_offset))));
+    }
+
+    // Give a better score to positions in the center
+    if (position.y() >= 0)
+    {
+        positionQuality =
+                positionQuality / (1 + std::exp(sigmoid_steepness * (position.y() - (width - y_offset))));
+    }
+    else if (position.y() < 0)
+    {
+        positionQuality =
+                positionQuality / (1 + std::exp(sigmoid_steepness * (-position.y() - (width - y_offset))));
+    }
+
```

Comment:
As a general note for this function, it would be a good idea to have a negative sigmoid inside the enemy's defense area, since we can't actually receive a ball in that area (The negative sigmoid would cancel out the positive ones in that area. Basically we want the score to drop to 0 inside the enemy defense area). We can pass through it, but the receiver point should not be inside the enemy defense area.

Adding a few sigmoids to cancel out the positive values in that area should do it.

Reply:
Roughly added that, will do once I get around to adding a circular sigmoid function